src/postgres/configurePostgresFirewall.ts

```diff
@@ -0,0 +1,20 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { GenericTreeItem, IActionContext } from "vscode-azureextensionui";
+import { ext } from "../extensionVariables";
+import { nonNullProp } from "../utils/nonNull";
+import { PostgresServerTreeItem } from "./tree/PostgresServerTreeItem";
+
+export async function configurePostgresFirewall(context: IActionContext, treeItem?: PostgresServerTreeItem | GenericTreeItem): Promise<void> {
+    if (!treeItem) {
+        treeItem = await ext.tree.showTreeItemPicker(PostgresServerTreeItem.contextValue, context);
+    } else if (treeItem instanceof GenericTreeItem) {
+        treeItem = nonNullProp(nonNullProp(treeItem, 'parent'), 'parent');
```

Comment:
nit: I would say this is the slightly cleaner way of getting around this problem: https://github.com/microsoft/vscode-azuretools/blob/b37737c0550d5e8af177228d33142d469f1a27ef/appservice/src/tree/LogFilesTreeItem.ts#L34

Aka set the commandArgs on the GenericTreeItem to the parent or whatever so that this file always gets a `PostgresServerTreeItem`